src/Aquifer.Common/Services/ResourceContentSearchService.cs

```diff
@@ -0,0 +1,672 @@
+﻿using Aquifer.Data;
+using Aquifer.Data.Entities;
+using Dapper;
+using Microsoft.EntityFrameworkCore;
+using static Aquifer.Common.Services.ResourceContentSearchResult;
+
+namespace Aquifer.Common.Services;
+
+public interface IResourceContentSearchService
```

Comment:
If it would be easier for review then I could do a pairing session to walk through this class.